src/scrapers/visa-cal.js

```diff
@@ -250,14 +253,9 @@ class VisaCalScraper extends BaseScraper {
         };
       }
     }
-
-    if (_.get(authResponse, 'Response.Status.Succeeded')) {
-      this.authHeader = `CalAuthScheme ${authResponse.AuthenticationToken}`;
-      this.emitProgress(SCRAPE_PROGRESS_TYPES.LOGIN_SUCCESS);
-      return { success: true };
-    }
-
-    throw new Error('unknown error during login');
+    this.authHeader = `CALAuthScheme ${authResponse.token}`;
+    this.emitProgress(SCRAPE_PROGRESS_TYPES.LOGIN_SUCCESS);
+    return { success: true };
```

Comment:
Isn't there a way to know if the authentication failed?
Because here it seems that it will always return success regardless of whether the authentication succeeded.

Reply:
The two blocks above check if Response.Status.Message exist in the payload, which appears to be the indication for a failure. Looking at it again now, I would add another generic error after line 254, that aborts if the "token" parameter does not exist and we couldn't match any of the more specific error messages.

Reply:
It will be a good guard making sure the 'token' exists